kernelci/build.py

```diff
@@ -381,14 +381,22 @@ def _download_file(url, dest_filename, chunk_size=1024):
     headers = {
         'User-Agent': 'kernelci {}'.format(kernelci_version),
     }
-    resp = requests.get(url, stream=True, headers=headers)
-    if resp.status_code == 200:
-        with open(dest_filename, 'wb') as out_file:
-            for chunk in resp.iter_content(chunk_size):
-                out_file.write(chunk)
-        return True
-    else:
-        return False
+    max_tries = 10
+    for i in range(max_tries):
+        try:
+            resp = requests.get(url, stream=True, headers=headers)
+            if resp.status_code == 200:
+                with open(dest_filename, 'wb') as out_file:
+                    for chunk in resp.iter_content(chunk_size):
+                        out_file.write(chunk)
+                    return True
+            break
+        except requests.exceptions.RequestException as e:
+            print(f'_download_file exception {e}, might retry')
+            time.sleep(2*i)
+            continue
```

Comment:
I think that this part can be improved a bit.

If it's the last iteration then the it'll sleep for 18 seconds and then exit the loop.
Maybe it can be change so the code is aware that it reached the maximum number of retries and exit immediately in case of a request failure. This can also make it possible to change the ambiguous message `"might retry"` to  something like `"retrying in <number> seconds"`

Btw, is it done on purpose to leave without any message if the request returned with HTTP status other than `200`?

Reply:
I think we could use a similar mechanism as implemented in `find_pods()` function to handle retries here

Reply:
I implemented printing information about retry delay. 

It is implemented a bit differently than find_pods(), because nature of errors are a bit different as well. With find_pods it is almost always situation with expired token or random/short network failure, which is often fixed from 2-3nd retry and dont need significant delay.
Here it might be situation that networking stack coming up too slow or there is congestion on storage, which implies we need some kind of backoff/increased delay on each retry to not make situation worse.
I will leave comment, why such design choice made.

I implemented retry if code is not 200 as well, you are absolutely right, i missed this point, it might be case with intermittent storage issues (nginx or backend reloading, etc)